tests/extrusion/test_rhs_side_bcs.py

```diff
@@ -21,7 +21,7 @@ def run_test(x, degree, quadrilateral, parameters={}, test_mode=False):
     for bc in bcs:
         bc.apply(u)
     v = Function(V)
-    v.interpolate(Expression("x[0] < 0.05 ? 10.0 : x[0] > 0.95 ? 42.0 : 0.0"))
+    v.interpolate(Expression("if(x[0] < 0.05, 10, if(x[0] > 0.95, 42, 0))"))
```

Comment:
OK, so `Expression` is now not C string, but loopy. This needs to be documented, or else we need to figure out a way to not do that.

Reply:
If feasible, I'd recommend not using strings to pass information to loopy. Instead, simply build the equivalent expression in pymbolic.

Reply:
This is "user-facing" code, rather than auto-generated, so I would prefer not to have users writing pymbolic expressions if possible. Previously it was for pasting raw C strings into kernels.

Reply:
We already have a deprecation policy on C strings. It's in the manual: https://www.firedrakeproject.org/interpolation.html#c-string-expressions

We only keep C string expressions for FEniCS compatibility (and if they are going to contain something that isn't actually C then we're not compatible and we should just drop them).

I am not aware of any live use cases for C string expressions that can't be done in UFL. If they do exist, then we should extend UFL to fix them. In extremis the user can fall back to a PyOP2 kernel.

Reply:
OK, so my suggestion, if we can't keep C string compatibility:

1. replace `Expression` with a function that raises an error with "how to use UFL expressions"
2. expunge `Expression` from everywhere in Firedrake.